python/paddle/distributed/auto_parallel/static/reshard_funcs/same_status_reshard_func.py

```diff
@@ -127,32 +120,37 @@ def reshard(self, src_dist_attr, dst_dist_attr, src_value, dst_type):
                     -1 not in dst_type.shape
                 ), "dynamic shape is not supported by pir-auto parallel yet."
 
-                comm_group = new_process_group([src, dst], group_type="p2p")
-                recv_value = paddle._C_ops.recv_v2(
-                    dst_type._local_shape,
-                    dst_type.dtype,
-                    comm_group.ranks.index(src),
-                    comm_group.id,
-                    True,
-                    False,
+                # create the buffer on other ranks for receving the data
+                value_dist_attr = copy_dist_attr_with_new_member(
+                    dst_dist_attr, new_process_mesh=dst_mesh
                 )
-                new_op = recv_value.get_defining_op()
-                new_op.dist_attr = (
+                tmp_value = paddle.zeros(dst_type._local_shape, dst_type.dtype)
+                tmp_value.set_type(dst_type)
+                op = tmp_value.get_defining_op()
+                op.dist_attr = (
                     paddle.base.libpaddle.pir.create_op_dist_attribute(
-                        dst_mesh,
-                        [],
-                        [dst_dist_attr],
-                        chunk_id,
+                        dst_mesh, [], [value_dist_attr], chunk_id
                     )
                 )
-                recv_value.set_type(dst_type)
-                is_send = False
-                break
 
+            comm_group = new_process_group([src, dst])
+            broadcast_value = paddle._C_ops.broadcast(
+                tmp_value,
+                comm_group.id,
+                comm_group.ranks.index(src),
+            )
+            broadcast_value.set_type(dst_type)
+            broadcast_op = broadcast_value.get_defining_op()
+            bcast_mesh = merge_process_meshes([src_mesh, dst_mesh])
+            broadcast_op.dist_attr = (
+                paddle.base.libpaddle.pir.create_op_dist_attribute(
+                    bcast_mesh, [src_dist_attr], [dst_dist_attr], chunk_id
+                )
+            )
         if is_send:
             # fake var will be removed in remove_other_rank_op_pass.
             fake_var = paddle._C_ops.reshard_v2(src_value, dst_dist_attr)
             fake_var.set_type(dst_type)
             return fake_var
```

Comment:
why still need fake_var?

Reply:
> why still need fake_var?

If the current rank is not src rank or dst rank, fakevar is needed.